vortex-scalar/src/pvalue.rs

```diff
@@ -38,28 +38,65 @@ impl PValue {
         }
     }
 
-    pub fn reinterpret_cast<T: NativePType + Into<PValue>>(&self) -> Self {
-        if T::PTYPE == self.ptype() {
+    pub fn reinterpret_cast(&self, ptype: PType) -> Self {
+        if ptype == self.ptype() {
             return *self;
         }
 
-        assert!(T::PTYPE.is_int(), "Can only reinterpret cast integers");
         assert_eq!(
-            T::PTYPE.byte_width(),
+            ptype.byte_width(),
             self.ptype().byte_width(),
             "Cannot reinterpret cast between types of different widths"
         );
 
         match self {
```

Comment:
this could maybe be macroed but not sure it saves us that much space/complexity

Reply:
Hm, there's a specific clippy rule that really wants you to not transumte floats to ints.

Reply:
I think you're right it would be simpler to read if it was transmute on every line